src/main/java/dev/jbang/cli/BaseScriptCommand.java

```diff
@@ -157,23 +156,39 @@ public static Script prepareScript(String scriptResource, List<String> arguments
 			s.setAdditionalDependencies(dependencies);
 			s.setAdditionalClasspaths(classpaths);
 
+			// Logic to get all //SOURCES recursively
 			List<Path> resolvedSourcePaths = new ArrayList<>();
+			// It's important to know which sources we already visited,
+			// because many files can declare the same source.
 			Set<String> visited = new HashSet<>();
-			LinkedList<String> sources = new LinkedList<>();
+			// List of array: [0] original source [1] source
+			List<String[]> sources = new ArrayList<>();
```

Comment:
really should have something similar to FileRef for this :)

Reply:
I analyzed to use FileRef:
  1. Util.collectSources would have to return `List<FileRef>` instead of `List<String>`
  2. But to create a FileRef I would need to create a Script
  3. And to create a Script I would have to create a ScriptResource (or maybe pass `null` in the Script constructor).

So I would need to create 3 objects (FileRef, Script, ScriptResource) only to get what I already have and then discard them.
This would only make sense if they were not discarded and used later somehow.